crates/matrix-sdk/src/encryption/recovery/mod.rs

```diff
@@ -0,0 +1,526 @@
+    ///         RecoveryState::Enabled => {
+    ///             println!("Recovery has been enabled");
+    ///         }
+    ///         _ => (),
+    ///     }
+    /// }
+    /// # anyhow::Ok(()) };
+    /// ```
+    pub fn state_stream(&self) -> impl Stream<Item = RecoveryState> {
+        self.client.inner.recovery_state.subscribe_reset()
+    }
+
+    /// Enable secret storage *and* backups.
+    ///
+    /// This method will create a new secret storage key and a new backup if one
+    /// doesn't already exist. It will then upload all the locally cached
+    /// secrets, including the backup recovery key, to the new secret store.
+    ///
+    /// This method will throw an error if a backup already exists on the
+    /// homeserver but this [`Client`] isn't connected to the existing backup.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # use matrix_sdk::{Client, encryption::recovery::EnableProgress};
+    /// # use url::Url;
+    /// # async {
+    /// # let homeserver = Url::parse("http://example.com")?;
+    /// # let client = Client::new(homeserver).await?;
+    /// use futures_util::StreamExt;
+    ///
+    /// let recovery = client.encryption().recovery();
+    ///
+    /// let enable = recovery
+    ///     .enable()
+    ///     .wait_for_backups_to_upload()
+    ///     .with_passphrase("my passphrase");
+    ///
+    /// let mut progress_stream = enable.subscribe_to_progress();
+    ///
+    /// tokio::spawn(async move {
+    ///     while let Some(update) = progress_stream.next().await {
+    ///         let Ok(update) = update else {
+    ///             panic!("Update to the enable progress lagged")
+    ///         };
+    ///
+    ///         match update {
+    ///             EnableProgress::CreatingBackup => {
+    ///                 println!("Creating a new backup");
+    ///             }
+    ///             EnableProgress::CreatingRecoveryKey => {
+    ///                 println!("Creating a new recovery key");
+    ///             }
+    ///             EnableProgress::Done { .. } => {
+    ///                 println!("Recovery has been enabled");
+    ///                 break;
+    ///             }
+    ///             _ => (),
+    ///         }
+    ///     }
+    /// });
+    ///
+    /// let recovery_key = enable.await?;
+    ///
+    /// # anyhow::Ok(()) };
+    /// ```
+    pub fn enable(&self) -> Enable<'_> {
+        Enable::new(self)
+    }
+
+    /// Create a new backup if one does not exist yet.
+    ///
+    /// This method will throw an error if a backup already exists on the
+    /// homeserver but this [`Client`] isn't connected to the existing backup.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # use matrix_sdk::{Client, encryption::backups::BackupState};
+    /// # use url::Url;
+    /// # async {
+    /// # let homeserver = Url::parse("http://example.com")?;
+    /// # let client = Client::new(homeserver).await?;
+    /// use futures_util::StreamExt;
```

Comment:
Doesn't seem to be used in this example.

Reply:
Fixed in 77444856aab8b321070293c3d661a4efa2f33082